### PR TITLE
BDD scenarios and steps for Upgrade Risk Prediction Inference service

### PR DESCRIPTION
**Pull request by joselsegura**
# Description

New feature files and step implementations to run BDD tests over the new Upgrade Risk Predicion Inference service.

Fixes #CCXDEV-9841

## Type of change

- New test feature file
- Non-breaking change in test steps implementation
- This change requires a documentation update
- Configuration update

## Testing steps
`./ccx_upgrade_risk_inference_tests.sh`

## Checklist
* [ ] Pylint passes for Python sources
* [ ] sources has been pre-processed by Black
* [ ] updated documentation wherever necessary

### CODE REVIEW

**Review by tisnik — APPROVED** [review]
LGTM

File: features/README.md [hunk 1]
```diff
@@ -507,3 +523,15 @@ Directory where feature files with scenarios and scenario outlines are stored.
 * Check if Kafka broker is running locally
 * Check if Kafka broker is running on expected port
 
+## `ccx-upgrades-inference/perform_inference.feature`
+
+* Check Inference Service response with no body is sent in the request
+* Check Inference Service response with an invalid body is used in the request
+* Check Inference Service response with an valid body with invalid data is used in the request
```
Comment (juandspy) [inline features/README.md:530]: "an" should be "a"

File: features/README.md [hunk 2]
```diff
@@ -507,3 +523,15 @@ Directory where feature files with scenarios and scenario outlines are stored.
...
+* Check Inference Service response with an valid body with valid data is used in the request
```
Comment (juandspy) [inline features/README.md:531]: "an" should be "a"

**Review by juandspy — APPROVED** [review]
Makes sense to me although I'd like to see some more cases. Probably this covers everything we know at the moment. So let's wait to have more code in place + knowing the corner cases of the model and rest of pieces before adding more BDD tests.

File: features/README.md (hunk 1 above)
Comment (joselsegura) [reply features/README.md]: Fixed, thank you!
